### PR TITLE
Etl 4593: add monitor events for hive-registration

### PR DESCRIPTION
**Pull request by tuGithub**

### CODE REVIEW

**Comment by tuGithub** [comment]
1. Event related changes in HiveMetaStoreBasedRegister.java and HiveMetaStoreEventHelper.java.
2. Current code, gobblin-core depend on gobblin-hive-registration, because MetricContext will be created using Instrumented which is in gobblin-core, this will cause the circular dependency
  Resolved it by 
  A: move gobblin.hive.HiveSerDeWrapper to gobblin-utility package
  B: BaseDataPublisherWithHiveRegistration.java from gobblin-core to gobblin-hive-registration
       HiveRegistrationPublisher.java  from gobblin-core to gobblin-hive-registration
3. Fixed one bug for SalesforceExtractor.java (complained by Findbug)

**Comment by tuGithub** [comment]
@chavdar , @ibuenros 
please review it when you have time

File: gobblin-hive-registration/src/main/java/gobblin/hive/metastore/HiveMetaStoreEventHelper.java [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+  public static final String SUCCESS_POSTFIX = "Succeed";
+  public static final String FAILED_POSTFIX = "Failed";
+  
+  public static final String DB_NAME = "DBName";
+  public static final String TABLE_NAME = "TableName";
+  public static final String ERROR_MESSAGE = "ErrorMessage";
+
+  // Path Registration
+  protected static void submitSuccessfulPathRegistration(EventSubmitter eventSubmitter, HiveSpec spec) {
+    eventSubmitter.submit("PathRegistration" + SUCCESS_POSTFIX,
```
Comment (ibuenros) [inline gobblin-hive-registration/src/main/java/gobblin/hive/metastore/HiveMetaStoreEventHelper.java]: Please make `"PathRegistration"` a constant (same for all other strings in this class)

File: gobblin-hive-registration/src/main/java/gobblin/hive/metastore/HiveMetaStoreEventHelper.java [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+  protected static void submitFailedPathRegistration(EventSubmitter eventSubmitter, HiveSpec spec, Exception error) {
+    eventSubmitter.submit("PathRegistration" + FAILED_POSTFIX,
+        getAdditionalMetadata(spec, Optional.<Exception> of(error)));
+  }
+
+  private static Map<String, String> getAdditionalMetadata(HiveSpec spec,
+      Optional<Exception> error) {
+    ImmutableMap.Builder<String, String> builder =
+        ImmutableMap.<String, String> builder().put(DB_NAME, spec.getTable().getDbName())
+            .put(TABLE_NAME, spec.getTable().getTableName()).put("Path", spec.getPath().toString());
```
Comment (ibuenros) [inline gobblin-hive-registration/src/main/java/gobblin/hive/metastore/HiveMetaStoreEventHelper.java]: If a partition is present in the spec you also want to add that information.

File: gobblin-hive-registration/src/main/java/gobblin/hive/publisher/BaseDataPublisherWithHiveRegistration.java [hunk 3]
```diff
@@ -10,13 +10,14 @@
  * CONDITIONS OF ANY KIND, either express or implied.
  */
 
-package gobblin.publisher;
+package gobblin.hive.publisher;
```
Comment (ibuenros) [inline gobblin-hive-registration/src/main/java/gobblin/hive/publisher/BaseDataPublisherWithHiveRegistration.java]: What's the purpose of the package change? Are we sure this won't break any external dependents?

File: gobblin-utility/build.gradle [hunk 4]
```diff
@@ -35,6 +35,9 @@ dependencies {
   compile externalDependency.commonsPool
   compile externalDependency.hadoopClientCommon
   compile externalDependency.avroMapredH2
+
+  compile externalDependency.hiveExec
```
Comment (ibuenros) [inline gobblin-utility/build.gradle]: Why are Hive dependencies required in gobblin-utility?

File: gobblin-utility/src/main/java/gobblin/util/hive/HiveSerDeWrapper.java [hunk 5]
```diff
@@ -10,7 +10,7 @@
  * CONDITIONS OF ANY KIND, either express or implied.
  */
 
-package gobblin.hive;
+package gobblin.util.hive;
```
Comment (chavdar) [inline gobblin-utility/src/main/java/gobblin/util/hive/HiveSerDeWrapper.java]: This is a backwards incompatible change that will break anyone who depends on HiveSerDeWrapper . Why is that necessary?

File: gobblin-salesforce/src/main/java/gobblin/salesforce/SalesforceExtractor.java [hunk 6]
```diff
@@ -627,6 +627,8 @@ public boolean bulkApiLogin() throws Exception {
 
       this.bulkConnection = new BulkConnection(config);
       success = true;
+    } catch (RuntimeException re) {
```
Comment (chavdar) [inline gobblin-salesforce/src/main/java/gobblin/salesforce/SalesforceExtractor.java]: This is a NOOP.

File: gobblin-hive-registration/src/main/java/gobblin/hive/metastore/HiveMetaStoreEventHelper.java [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+
+    if (error.isPresent()) {
+      builder.put(ERROR_MESSAGE, error.get().getMessage());
+    }
+
+    return builder.build();
+  }
+
+  // DB Creation
+  protected static void submitSuccessfulDBCreation(EventSubmitter eventSubmitter, String dbName) {
```
Comment (chavdar) [inline gobblin-hive-registration/src/main/java/gobblin/hive/metastore/HiveMetaStoreEventHelper.java]: Can you please add a github wiki documenting the event types and metadata ?

**Comment by tuGithub** [comment]
replaced with https://github.com/linkedin/gobblin/pull/1128
